.github/workflows/ci.yml

```
@@ -1,48 +1,36 @@
-      run: pytest -v --cov=gameinsights tests/
-    
+    - name: Install uv
+      uses: astral-sh/setup-uv@v4
+    - run: uv sync --group dev --extra dataframe
+    - run: uv run ruff check gameinsights/ tests/ --output-format=github
+    - run: uv run black --check gameinsights/ tests/
+    - run: uv run mypy gameinsights/
+    - run: uv run pytest -vv --cov=gameinsights tests/
     - name: Upload coverage
       if: matrix.python-version == '3.12'
       uses: codecov/codecov-action@v3
```

actionlint (1.7.12)

[error] the runner of "codecov/codecov-action@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)